src/shared/sam/cli/samCliDeploy.ts

```diff
@@ -8,55 +8,53 @@
 import { map } from '../../utilities/collectionUtils'
-import { DefaultSamCliProcessInvoker } from './samCliInvoker'
 import { SamCliProcessInvoker } from './samCliInvokerUtils'
 
-export class SamCliDeployInvocation {
-    public constructor(
-        private readonly templateFile: string,
-        private readonly stackName: string,
-        private readonly region: string,
-        private readonly parameterOverrides: Map<string, string>,
-        private readonly invoker: SamCliProcessInvoker = new DefaultSamCliProcessInvoker(),
-        private readonly profile: string
-    ) {}
-
-    public async execute(): Promise<void> {
-        const logger: Logger = getLogger()
-
-        const args = [
-            'deploy',
-            '--template-file', this.templateFile,
-            '--stack-name', this.stackName,
-            '--capabilities', 'CAPABILITY_IAM',
-            '--region', this.region,
-            '--profile', this.profile
+export interface SamCliDeployArgs {
+    templateFile: string
+    parameterOverrides: Map<string, string>
+    profile: string
+    region: string
+    stackName: string
+}
+
+export async function runSamCliDeploy(
+    deployArguments: SamCliDeployArgs,
+    invoker: SamCliProcessInvoker,
+    logger: Logger = getLogger(),
+): Promise<void> {
+    const args = [
+        'deploy',
+        '--template-file', deployArguments.templateFile,
+        '--stack-name', deployArguments.stackName,
+        '--capabilities', 'CAPABILITY_IAM',
+        '--region', deployArguments.region,
+        '--profile', deployArguments.profile
+    ]
+    if (deployArguments.parameterOverrides.size > 0) {
+        const overrides = [
+            ...map(
+                deployArguments.parameterOverrides.entries(),
+                ([key, value]) => `${key}=${value}`
+            )
         ]
-        if (this.parameterOverrides.size > 0) {
-            const overrides = [
-                ...map(
-                    this.parameterOverrides.entries(),
-                    ([key, value]) => `${key}=${value}`
-                )
-            ]
-            args.push('--parameter-overrides', ...overrides)
-        }
-
-        const { exitCode, error, stderr, stdout }: ChildProcessResult = await this.invoker.invoke(...args)
-
-        if (exitCode === 0) {
-            return
-        }
-
-        console.error('SAM deploy error')
-        console.error(`Exit code: ${exitCode}`)
-        console.error(`Error: ${error}`)
-        console.error(`stderr: ${stderr}`)
-        console.error(`stdout: ${stdout}`)
-
-        const message = error && error.message ? error.message : stderr || stdout
-        const err = new Error(`sam deploy encountered an error: ${message}`)
-        logger.error(err)
-        throw err
+        args.push('--parameter-overrides', ...overrides)
     }
+
+    const { exitCode, error, stderr, stdout }: ChildProcessResult = await invoker.invoke(...args)
```

Comment:
We should also movie this boilerplate error checking to a common location. Can be in a separate PR if it's out-of-scope for this PR.

Reply:
It's like you're reading my mind! 
After this PR I am making a similar change to sam package, and then I planned to unify the error checking in a separate PR (to avoid PR collisions and bigger PRs)